cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -35,23 +35,48 @@ function main {
     exit 1
   fi
 
+  echo "Waiting for $device to come online"
+  until test -b "$device"; do sleep 1; done
   partition=${device}1
   if [[ ! -b "$partition" ]]
   then
     echo "Partition $partition not detected: creating partitions"
     parted -s -a optimal "$device" -- \
       mklabel gpt \
       mkpart primary xfs 1 -1
-    partprobe "$device"
+    partprobe "$device" > /dev/null 2>&1 || :
+    echo "Waiting for $partition to be detected by the Kernel"
+    until test -b "$partition"; do sleep 1; done
     echo "Formatting: $partition"
-    mkfs.xfs "$partition" >/dev/null
+    mkfs.xfs "$partition" > /dev/null
     echo "Setting up partition mount"
     mkdir -p "$mount_location"
-    fstab="$partition $mount_location xfs defaults,nofail 0 2"
+    fstab="$partition $mount_location xfs defaults 0 2"
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
-    echo "Mounting: $partition to $mount_location"
-    mount -a
+    if [ "$mount_location" = "/var/log" ]; then
```

Comment:
Let's add some details to the usage section about this special case. We should call out that when the script is called with a mount location of `/var/log` the contents will be migrated and certain services will be restarted to accomplish the migration.